packages/painperdu/src/PainPerduListItem/PainPerduListItem.tsx

```diff
@@ -0,0 +1,60 @@
+import { useState } from 'react';
+import type { FC } from 'react'
+import type { PathItem } from '../types';
+import '../index.css'
+
+interface Props {
+  pathItem: PathItem[],
+}
+
+export const PainPerduListItem: FC<Props> = ({ pathItem }) => {
+  const [selectedPath, setSelectedPath] = useState<string>('')
+  const [pathClass, setPathClass] = useState<string>('')
+
+ const getSelectedPath = (path: PathItem, index: number): void => {
+  setSelectedPath(path.alias)
+
+  if (selectedPath === path.alias) {
+    pathItem[index].isSelected = true
```

Comment:
Il ne faut pas directement update les props, une des techniques que tu peux faire c'est, sachant que les datas viennent du parents tout ce qui est update des datas il faut que ce soit le parent qui le gère et qui du coup va rerendre les composants enfants avec la nouvelle donnée

Reply:
Alors oui mais il y aussi la technique d'utiliser une current data dans le composant enfant et à l'init du compo de l'init à la props, comme ça on garde la logique dans le compo enfant. 
Ca évite de surcharger le compo parent avec encore plus de logique, sachant qu'il va y en avoir beaucoup

Reply:
Hmm oui mais tu gardes le principes de single responsabilities en gros le composant listItem c'est un composant générique qui s'occupe de display sachant que la data vient du parent c'est aux parent à gérer tout ce qui est modificiation de données et qui pour le coup va rerendre les enfants et c'est un peu la convention:

https://stackoverflow.com/questions/24939623/can-i-update-a-components-props-in-react-js
https://www.freecodecamp.org/news/how-to-update-a-components-prop-in-react-js-oh-yes-it-s-possible-f9d26f1c4c6d/

Reply:
Alors je suis d'accord mais je pense que c'est quelque chose qui reste à challenger ou à en discuter mais pour avoir une réponse et solution rapidement : on va gérer la logique dans le parent qui passera la data à l'enfant et dans l'enfant je vais laisser la props qui donnera les bonnes data

Reply:
Si tu veux on peut challenger le truc mais c'est un peu une sorte de convention quand t'as des props que tu veux updater tu le fais dans le composant c'est pas son taff mais à voir si tu veux

Reply:
Déjà updater une props ce n'est pas la même des choses à faire mais j'ai l'impression qu'en React c'est plus... flex que en Vue où là c'est complètement interdit, warnings et errors de partout..

Reply:
Hmm je pense que en vue et react t'as des erreurs je suis quasi sur qu'en React comme ils utilisent https://immutable-js.com/ tu ne peux pas venir update une props donc tu vas devoir spread ton objet props donc si tu as par exemple 100 objects dans ton parent tu vas devoir spread ton object également dans ton enfant et donc te retrouver avec 200 objects splité entre ton parent et ton enfant

Reply:
Bah je n'ai eu aucune error pour cet update de props, donc c'est pour ça que j'étais surpit..
Mais la finalité du truc : c'est que la logique va être traitée dans le parent et l'enfant va juste recevoir, en props, les data du parent et afficher

Reply:
Non il va dire la emit à mon parent ça emit ça et le parent va dire ok donc j'update cet enfant tiens voila la nouvelle donnée

Reply:
Ah oui oui, autant pour moi ! 
Emit pour avoir les data, yes je suis d'accord

Reply:
Comme ça le parent lui son taff sachant que c'est un composant générique il prend des props et dit ok lui il se passe ça est-ce que tu peux me donner les nouvelles données ? Le parent va écouter il va dire ok je fais le traitement tiens voila la nouvelle donnée du coup j'update mon DOM comme ça chacun à sa responsabilité et aussi question perf t'es obligé de dupliquer tes props si tu veux faire toute la logique dans ton enfant et donc te retrouver avec potentiellement deux fois plus de datas à store dans ton app.

En fait ton parent dans le return il doit pas avoir beaucoup de DOM lui c'est un composant qui va gérer de la datas et la transmettre aux enfants qui vont dire il se passe quoi tiens modifie moi ça etc..

Reply:
Corrected : [b3052bf](https://github.com/luctst/painperdu/pull/12/commits/b3052bf28a7f41285250c1ebbe97aa89d1706c13)